### ISSUE TITLE
About a Pypi Wheel

### ISSUE THREAD

**Issue by KelSolaar** [AcademySoftwareFoundation/OpenColorIO#1362]
Hi,

Just kicking out the discussion for this one! With OpenColorIO 2.x.x released, it would be great to have a Pypi wheel available, that would make integration with the Python ecosystem orders-of-magnitude easier. The OpenColorIO-Config-ACES repo could use it right-of-the bat.

@michdolan has started some work on that, so maybe we can continue from there?

Cheers,

Thomas

**Comment by remia** [comment]
Hey @michdolan,

Was curious what was your take on this ?

I started looking into it and it seems to me that using an external builder like [cibuildwheel](https://github.com/joerick/cibuildwheel) would be a good idea to handle the build environment for us, it's used in many project, as listed in the Readme, including[ pybind11's example repository](https://github.com/pybind/cmake_example).

**Comment by michdolan** [comment]
I think that's a great idea. Looks like an awesome tool! I'm definitely on board with not reinventing the wheel (no pun intended) and we use GitHub Actions, so should be doable.

**Comment by remia** [comment]
Testing this workflow, I uploaded some wheel here https://test.pypi.org/project/opencolorio/ and the code can be seen here https://github.com/remia/OpenColorIO/pull/8.

Would be interested in any feedback and/or tests, thanks.

**Comment by zachlewis** [comment]
Thanks @remia  -- works like a charm on CentOS-7.4 (python-3.7.4) / MacOS-11.2 (python-3.8.2) / whatever Google Colab uses (python-3.7). Nice work!

**Comment by remia** [comment]
Thanks for testing @zachlewis. I realised that the version number is incorrect as it's built from the master branch, I will fix that, among other improvements.

**Comment by remia** [comment]
Should I open a PR here to start review / discussion now that the basis workflow is in place ? The upload to https://test.pypi.org/ is obviously relying on GitHub repository secret but I guess it doesn't really matter at first when reviewing the code.

**Comment by zachlewis** [comment]
Please do! I think this will be really well-received. Much appreciated!

**Comment by remia** [comment]
Just wanted to note there was mention about adding a conda-forge package on slack: https://conda-forge.org/#about

It could be done after the PyPI release and would the package recipe would live in it's own repo within the conda-forge github org from what I understand.

**Comment by KelSolaar** [comment]
Yes correct, it is almost automatic and rather low effort to maintain.
